builtin/providers/vsphere/resource_vsphere_virtual_disk.go

```diff
@@ -0,0 +1,250 @@
+package vsphere
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+type virtualDisk struct {
+	size        int
+	vmdkPath    string
+	initType    string
+	adapterType string
+	datacenter  string
+	datastore   string
+}
+
+// Define VirtualDisk args
+func resourceVSphereVirtualDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereVirtualDiskCreate,
+		Read:   resourceVSphereVirtualDiskRead,
+		Delete: resourceVSphereVirtualDiskDelete,
+
+		Schema: map[string]*schema.Schema{
+			// Size in GB
+			"size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true, //TODO Can this be optional (resize)?
```

Comment:
That would be changed in a PR that also adds vm Update functionality.  For now it needs to stay this way, else TF would go looking for a `resourceVSphereVirtualMachineUpdate` that isn't there.